package.json

```diff
@@ -21,7 +21,12 @@
     "analyze": "npm run clean && npm run build:prod -- --stats-json && webpack-bundle-analyzer ./dist/stats.json",
     "contributors:add": "all-contributors add",
     "contributors:generate": "all-contributors generate && node .all-contributors-html.js",
-    "contributors:check": "all-contributors check"
+    "contributors:check": "all-contributors check",
+    "cy:open": "cypress open",
+    "cy:run": "cypress run",
+    "precy:ci": "npm run build",
+    "cy:ci": "npm-run-all --parallel --race cy:server \"cy:run -- --config=baseUrl=http://localhost:4000\"",
```

Comment:
Do I understand correctly that this tries to check if there is already existing prod build in `dist` so that it can skip rebuilding but will rebuild with `cy:run` if `./dist` is empty ?

Reply:
No this will always build the app, once it's build it will start a server on port 4000 which is used by cypress to test the app.

Reply:
Hmm, I am just trying to wrap my head around that the `npm-run-all` with

> `--race`  - Set the flag to kill all tasks when a task **finished with zero**. This option is valid only with 'parallel' option.

and

> Well-behaved UNIX commands, programs, and utilities **return a 0** exit code upon successful completion, though there are some exceptions.

So then only one of them (the faster will complete) ? Why do we try both ?

Reply:
If we wouldn't do this and simply do "npm run cy:server && npm run cy:run", the `cy:run` command will never be run because the server command will not return an exit code (it just sits there and waits).

With `npm-run-all` you can run multiple jobs in parallel, so in our case both the `server` and `run` command will execute. Like I said before the `server` command just sits there, but the `cy:run` command will return an exit code when all the tests have run. When it retrieves an exit code it will also shut down the server.

Reply:
Ah ok, I understand now, it's a way to tear down server when the tests are finished. Clever! 

I guess I got confused because I was not aware of expected lifetimes of those commands, now it all make sense. Maybe we can also add a small `Cypress testing` section in the `README.md` with the happy path usage scenario and gotchas ;)